test/unit/lib/plugins/package/lib/packageService.test.js

```diff
@@ -566,31 +198,37 @@ describe('lib/plugins/package/lib/packageService.test.js', () => {
           },
         },
       });
+
+      fnIndividualZippedFiles = await listZipFiles(
+        path.join(servicePath, '.serverless', 'fnIndividual.zip')
+      );
     });
 
     it('should exclude custom plugins localPath', () => {
-      // Confirm ".serverless-plugins/index.js" is not packaged
+      expect(fnIndividualZippedFiles).to.not.include('.custom-plugins/index.js');
     });
 
-    it('should ignore `package.artifact` if `package.individually`', () => {
-      // Replace
+    it.skip('TODO: should ignore `package.artifact` if `package.individually`', () => {
```

Comment:
Is there a reason to why it's still kept skipped? Would be good to implement it as the original test is being removed as a part of this PR

Reply:
Sorry, I haven't figured it out how to test it properly, since the current one doesn't work as I'd expect.

Reply:
After thinking a bit about this one I also don't see a clear path to provide a reasonable replacement here, @medikoo do you have any suggestions?

Reply:
It's to confirm that `package.artifact` is effectless with `package.individually`.

Do you see it not happen? I think if there's a `package.individually` then global `package.artifact` is always ignored and each function is packaged independently. Is that not the case?

Reply:
I skipped this test because `artifact.zip` was found in individually packaged bundle. I thought that's my misunderstanding but I wasn't able to find a root cause or a correct solution here.

Reply:
@medikoo That's all true, but how would you go about asserting that in this test case? I have a hard time constructing a meaningful assertion here

Reply:
@scadu I believe it ends up in the individually packaged bundle because it's in fixture's directory and it's not explicitly exluded anywhere during packaging, hence also being gathered during packaging of that individual function

Reply:
packaging logic is expected to generate artifact for function individually. When it's created it's assigned to `functions[].package.artifact` (and then deploy upload logic reads it from there to upload it to S3)

So I think proper test is to retrueve given function configuration from `serverless` instance and confirm that it's `package.artifact` is set and doesn't reflect service wide `package.artifact`

Reply:
@medikoo makes sense now. I'll look into it.

Reply:
I tried covering this one in https://github.com/serverless/serverless/pull/8721/commits/daf13009e47446bf06bf7c32e4f4ad0bd1d2606d and still only partially (assuming that the assertion is ok). I must admit I struggle a little bit here and I'm sorry for dragging this MR out.
Seems like I still miss a few bits here and there to fully refactor `packageService` test.

Reply:
@scadu I believe you're on the right track with that implementation :100: In that assertion, we should check that the `package.artifact` on function exists and that it's different than the one set on service wide `package.artifact` and we should be good to go with this particular test :raised_hands:

Reply:
@pgrzesik thanks for confirming. I just wonder how to pass properties from the `system wide` test to `individually` without nesting or something, but I'll go through mocha's documentation once again to find some pointers.

Reply:
I don't think you need to pass properties between these tests - in current test you can access the artifact set on the function and confirm that it's different that the one from `package.artifact` in config

Reply:
I think it should be fixed now. All cases are now covered.